packages/devtools-source-map/src/utils/getOriginalStackFrames.js

```diff
@@ -3,8 +3,123 @@
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
 // @flow
+/* eslint camelcase: 0*/
 
-import type { Location } from "debugger-html";
+import type { Location, SourceId } from "debugger-html";
+
+const { getSourceMap } = require("./sourceMapRequests");
+const { generatedToOriginalId } = require("./index");
+
+const xScopes = new Map();
+
+type XScopeItem = any;
+
+function indexLinkingNames(items: XScopeItem[]): Map<string, XScopeItem> {
+  const result = new Map();
+  let queue = [...items];
+  while (queue.length > 0) {
+    const item = queue.shift();
+    if ("linkage_name" in item) {
+      result.set(item.linkage_name, item);
+    }
+    if ("children" in item) {
+      queue = [...queue, ...item.children];
+    }
+  }
+  return result;
+}
+
+async function getXScopes(sourceId: SourceId): Promise<*> {
```

Comment:
can we move `getXScopes` and the filter logic (maybe `formatXScopes`) to a separate module. Also on the debugger side, is it better to describe the scopes as WasmScopes as opposed to XScopes, i assume X comes from the sourcemap extension language, but I think it is confusing to others who might think XScopes is something non-wasm... and special if we dont namespace it